### PR TITLE
[DPE-3232] Apache ZooKeeper <> OCI Factory Integration

### PR DESCRIPTION
**Pull request by deusebio**
- [x] I have signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] I am following the [CONTRIBUTING](https://github.com/canonical/oci-factory/blob/main/CONTRIBUTING.md) guidelines

*Ping the @canonical/rocks team.*

---

## Description
We have recently created a Rock for packaging Apache ZooKeeper [here](https://github.com/canonical/zookeeper-rock/pull/1). The binaries are the same as the ones used in Charmed ZooKeeper images, but here we just reduce only to the Apache ZooKeeper, leaving outside all the extra components and dependencies. 

The idea of the image would be to replace the current Apache ZooKeeper images in DockerHub that are not integrated with Rock and OCI Factory pipeline.

### CODE REVIEW

File: oci/zookeeper/documentation.yaml [hunk 1]
```diff
@@ -18,6 +18,41 @@ description: |
...
+    ### Debugging
+    
+    To debug the container:
+    ```bash
+    docker logs -f <zookeeper-container>
+    ```
+    
+    To get an interactive shell:
+    ```bash
+    docker exec -it <zookeeper-container> /bin/bash
```
Comment (deusebio) [inline oci/zookeeper/documentation.yaml:57]: @cjdcordeiro I tried to have this consistent with [this](https://github.com/canonical/oci-factory/pull/48/files#diff-9bb32735cb189a38cdbe3ee4c131c0a2d970adfa2c35290ebb9e478d480e7e98R49), but the Rock syntax

```
docker exec -it <zookeeper-container> exec /bin/bash 
```

did not work for me. It provided this output:

```
CI runtime exec failed: exec failed: unable to start container process: exec: "exec": executable file not found in $PATH: unknown
```

I believe it is expected, but I'm not sure why it works for the other image.

Comment (cjdcordeiro) [reply oci/zookeeper/documentation.yaml:57]: hum weird. It works for me.

I took your rock from https://github.com/canonical/zookeeper-rock/actions/runs/8094163188, 

then `skopeo copy oci-archive:zookeeper_3.8.2_amd64.rock docker-daemon:test:latest` and then

```bash
$ docker run --rm -it test exec /bin/bash
2024-02-29T15:14:03.595Z [pebble] Started daemon.
2024-02-29T15:14:03.601Z [pebble] POST /v1/exec 3.14273ms 202
2024-02-29T15:14:03.604Z [pebble] GET /v1/tasks/1/websocket/control 2.84456ms 200
2024-02-29T15:14:03.604Z [pebble] GET /v1/tasks/1/websocket/stdio 40.835µs 200
root@ae43a10f8a7c:~# 
root@ae43a10f8a7c:~# 
root@ae43a10f8a7c:~# whoami
root
root@ae43a10f8a7c:~# 
```

can you share your steps?

Comment (deusebio) [reply oci/zookeeper/documentation.yaml:57]: yes, but that's not what is in the documentation I linked. If you do:

```
docker run --rm zookeeper:latest
```

you start the zookeeper process. But you then want to debug something that requires to login in the container. Then I first find the container id with `docker ps`, and use the container id in the command

```
docker exec -it <container-id> exec /bin/bash
```

This throws an error 

```
OCI runtime exec failed: exec failed: unable to start container process: exec: "exec": executable file not found in $PATH: unknown
```

On the other hand, using the syntax

```
docker exec -it <container-id> /bin/bash
```

works fine.

**Comment by deusebio** [comment]
Hi Cris! I believe that the PR is now ready for review! All checks pass (also the security scanning now that we have released a new version of the artifacts)

**Comment by cjdcordeiro** [comment]
greattt! thanks

**Review by cjdcordeiro — APPROVED** [review]

**Comment by deusebio** [comment]
wow! the bot works! I knew you merged the PR because I have received the MM ping! :D

**Pull request merged**